routes/event_signup_router.py

```diff
@@ -16,14 +16,52 @@
 
 # Sing current user up to an event
 @event_signup_router.post("/{event_id}", response_model=EventRead)
-def signup_route(event_id: int, signup: EventSignupCreate, me: Annotated[User_DB, Permission.member()], db: DB_dependency):
+def signup_route(
+    event_id: int,
+    signup: EventSignupCreate,
+    me: Annotated[User_DB, Permission.member()],
+    manage_permission: Annotated[bool, Permission.check("manage", "Event")],
+    db: DB_dependency,
+):
     event = db.query(Event_DB).filter_by(id=event_id).one_or_none()
     if event is None:
         raise HTTPException(status.HTTP_404_NOT_FOUND)
 
-    signup_to_event(event, me, signup, db)
+    signup_to_event(event, me, signup, manage_permission, db)
     return event
 
+
+@event_signup_router.delete("/{event_id}", response_model=EventRead)
+def signoff_route(
+    event_id: int,
+    me: Annotated[User_DB, Permission.member()],
+    manage_permission: Annotated[bool, Permission.check("manage", "Event")],
+    db: DB_dependency,
+):
+    event = db.query(Event_DB).filter_by(id=event_id).one_or_none()
+    if event is None:
+        raise HTTPException(status.HTTP_404_NOT_FOUND)
+
+    signoff_from_event(event, me, manage_permission, db)
+    return event

```

Comment:
Return the deleted signup instead. A route generally returns the resource that was modified/deleted/created